once-per-save.lua

```diff
@@ -0,0 +1,57 @@
+-- runs dfhack commands unless ran already in this save
+
+local HELP = [====[
+
+once-per-save
+=============
+Runs commands like ``multicmd``, but only unless
+not already ran once in current save.
+
+Use this in ``onMapLoad.init`` with f.e. ``ban-cooking``::
+
+  once-per-save ban-cooking tallow; ban-cooking honey; ban-cooking oil; ban-cooking seeds; ban-cooking brew; ban-cooking fruit; ban-cooking mill; ban-cooking thread; ban-cooking milk;
+
+Only successfully ran commands are saved.
+
+Parameters:
+
+--help            display this help
+--rerun commands  ignore saved commands
+--reset           deleted saved commands
+
+]====]
+
+local STORAGEKEY = 'once-per-save'
+
+local args = {...}
+local rerun = false
+if args[1] == "--help" or args[1] == "-?" or args[1] == "?" then
```

Comment:
A lot of our up-to-date Lua scripts use `utils.processArgs()` - for a consistent experience, it might be better to use that.

Reply:
The problem I see with `utils.processArgs()` is we wouldn't be able to call `once-per-save ban-cooking tallow` anymore, only `once-per-save -command [ ban-cooking tallow ]`. I don't like added verbosity in this case.

Reply:
Ah, good point. I do think supporting arguments prefixed with a single `-` would be good, though. (I'm personally a fan of `--`, but I'd rather have DFHack scripts be consistent.)

Reply:
added single-dash alternatives, but I don't know how do I make reStructuredText generated by sphinx behave nicely in this case, so I kept double dashes in parameter list.